otter/test/convergence/test_gathering.py

```diff
@@ -213,85 +221,87 @@ class GetScalingGroupServersTests(SynchronousTestCase):
     """
 
     def setUp(self):
-        """Save basic reused data."""
```

Comment:
I'm not sure we should throw away the tests for `get_all_scaling_group_servers` - it's invoked by `get_scaling_group_servers`, which you now have tests for, but `get_all_scaling_group_servers` is stubbed out in those tests.

If the functionality for actually filtering stuff out by group, etc. in `get_all_scaling_group_servers` is broken, I'm not sure the unit tests would catch them now, and then `get_scaling_group_servers` would break because it assumes that `get_all_scaling_group_servers` would do it correctly.